backend/compact-connect/common_constructs/user_pool.py

```diff
@@ -107,11 +117,8 @@ def add_ui_client(self, callback_urls: List[str], ui_scopes: List[OAuthScope] =
             enable_token_revocation=True,
             generate_secret=False,
             refresh_token_validity=Duration.days(30),
-            # If you provide no attributes at all here, it will default
-            # to making _all_ attributes writeable, so if we want to limit writes,
-            # we have to provide at least _one_ that the client _can_ write.
-            write_attributes=ClientAttributes().with_standard_attributes(email=True),
```

Comment:
Nit: Can we ensure we don't run into this issue in the future, e.g. by asserting that `write_attributes` (in the new code) is non-empty?

Reply:
Now that we are adding type hints for these parameters, the developer should not be able to enter a None type for either the read or write attributes, so this issue should not occur again. The only time that CDK will default to including all attributes is if you pass in None for the argument.

Reply:
@jusdino corrected me on this section, it appears that even if the developer enters `write_attributes=ClientAttributes().with_standard_attributes()` all the attributes would be set to writable, which doesn't align with what the documentation says for that class 😞. I have added some cdk tests that explicitly check the read and write attributes that are set for the two user pool app clients, which would catch if someone attempts to change it to an empty list. Other than that, I'm not sure how you would make an assertion on the object itself to see what parameters it was called with.

Reply:
Those tests sound great- thanks for doing that! And ditto on the sad face ... what an API :-)